<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-gb"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Generate multi-omic differential networks — get_diffNetworks • multiDEGGs</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Generate multi-omic differential networks — get_diffNetworks"><meta name="description" content="Generate a multi-layer differential network with interaction p values"><meta property="og:description" content="Generate a multi-layer differential network with interaction p values"><meta property="og:image" content="https://elisabettasciacca.github.io/multiDEGGs/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">multiDEGGs</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/multiDEGGs_vignette.html">Differential Network Analysis with multiDEGGs</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/elisabettasciacca/multiDEGGs/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Generate multi-omic differential networks</h1>
      <small class="dont-index">Source: <a href="https://github.com/elisabettasciacca/multiDEGGs/blob/main/R/core_functions.R" class="external-link"><code>R/core_functions.R</code></a></small>
      <div class="d-none name"><code>get_diffNetworks.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Generate a multi-layer differential network with interaction p values</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_diffNetworks</span><span class="op">(</span></span>
<span>  <span class="va">assayData</span>,</span>
<span>  <span class="va">metadata</span>,</span>
<span>  category_variable <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span>
<span>  category_subset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  network <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  percentile_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.35</span>, <span class="fl">0.98</span>, by <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>,</span>
<span>  padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>  show_progressBar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cores <span class="op">=</span> <span class="fu">parallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/parallel/detectCores.html" class="external-link">detectCores</a></span><span class="op">(</span><span class="op">)</span><span class="op">/</span><span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-assaydata">assayData<a class="anchor" aria-label="anchor" href="#arg-assaydata"></a></dt>
<dd><p>a matrix or data.frame (or list of matrices or data.frames
for multi-omic analysis) containing normalised assay data.
Sample IDs must be in columns and probe IDs (genes, proteins...) in rows.
For multi omic analysis, it is highly recommended to use a named list of
data.
If unnamed, sequential names (assayData1, assayData2, etc.) will be
assigned to identify each matrix or data.frame.</p></dd>


<dt id="arg-metadata">metadata<a class="anchor" aria-label="anchor" href="#arg-metadata"></a></dt>
<dd><p>a named vector, matrix, or data.frame containing sample
annotations or categories. If matrix or data.frame, each row should
correspond to a sample, with columns representing different sample
characteristics (e.g., treatment group, condition, time point). The colname
of the sample characteristic to be used for differential analysis must be
specified in <code>category_variable</code>. Rownames must match the sample IDs used in
assayData.
If named vector, each element must correspond to a sample characteristic to
be used for differential analysis, and names must match sample IDs used in
the colnames of <code>assayData</code>.
Continuous variables are not allowed.</p></dd>


<dt id="arg-category-variable">category_variable<a class="anchor" aria-label="anchor" href="#arg-category-variable"></a></dt>
<dd><p>when metadata is a matrix or data.frame this is the
column name of <code>metadata</code> that contains the sample annotations to be used for
differential analysis</p></dd>


<dt id="arg-regression-method">regression_method<a class="anchor" aria-label="anchor" href="#arg-regression-method"></a></dt>
<dd><p>whether to use robust linear modelling to calculate
link p values. Options are 'lm' (default) or 'rlm'. The lm implementation
is faster and lighter.</p></dd>


<dt id="arg-category-subset">category_subset<a class="anchor" aria-label="anchor" href="#arg-category-subset"></a></dt>
<dd><p>optional character vector indicating a subset of
categories from the category variable. If not specified, all categories in
<code>category_variable</code> will be used.</p></dd>


<dt id="arg-network">network<a class="anchor" aria-label="anchor" href="#arg-network"></a></dt>
<dd><p>network of biological interactions provided by the user. The
network must be provided in the form of a table of class data.frame with only
two columns named "from" and "to".
If NULL (default) a network of 10,537 molecular interactions obtained from
KEGG, mirTARbase, miRecords and transmiR will be used.
This has been obtained via the <code>exportgraph</code> function of the MITHrIL tool
(Alaimo et al., 2016).</p></dd>


<dt id="arg-percentile-vector">percentile_vector<a class="anchor" aria-label="anchor" href="#arg-percentile-vector"></a></dt>
<dd><p>a numeric vector specifying the percentiles to be
used in the percolation analysis. By default, it is defined as
<code>seq(0.35, 0.98, by = 0.05)</code>, which generates a sequence of percentiles
starting at 0.35, meaning that targets (genes/proteins...) whose expression
value is under the 35th percentile of the whole matrix will be excluded.
This threshold can be modified by specifying a different starting point for
<code>seq</code>. For a more granular percolation analysis an higher optimisation of
the algorithm, <code>by = 0.05</code> can be modified in favour of lower values, but
this will increase the computational time.</p></dd>


<dt id="arg-padj-method">padj_method<a class="anchor" aria-label="anchor" href="#arg-padj-method"></a></dt>
<dd><p>a character string indicating the p values correction
method for multiple test adjustment. It can be either one of the methods
provided by the <code>p.adjust</code> function from <code>stats</code> (bonferroni, BH, hochberg,
etc.) or "q.value" for Storey's q values, or "none" for unadjusted p values.
When using "q.value" the <code>qvalue</code> package must be installed first.</p></dd>


<dt id="arg-show-progressbar">show_progressBar<a class="anchor" aria-label="anchor" href="#arg-show-progressbar"></a></dt>
<dd><p>logical. Whether to display a progress bar during
execution. Default is TRUE.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>logical. Whether to print detailed output messages during
processing. Default is TRUE</p></dd>


<dt id="arg-cores">cores<a class="anchor" aria-label="anchor" href="#arg-cores"></a></dt>
<dd><p>number of cores to use for parallelisation.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a <code>deggs</code> object containing differential networks incorporating
p values or adjusted p values for each link.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Single omic analysis:</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_metadata"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_rnaseqData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">deggs_object</span> <span class="op">&lt;-</span> <span class="fu">get_diffNetworks</span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">synthetic_rnaseqData</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span></span>
<span class="r-in"><span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span></span>
<span class="r-in"><span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span></span>
<span class="r-in"><span>                                 padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span></span>
<span class="r-in"><span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span> </span></span>
<span class="r-in"><span><span class="co"># multi-omic analysis: </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_metadata"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_rnaseqData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_proteomicData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_OlinkData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">assayData_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"RNAseq"</span> <span class="op">=</span> <span class="va">synthetic_rnaseqData</span>,</span></span>
<span class="r-in"><span>                       <span class="st">"Proteomics"</span> <span class="op">=</span> <span class="va">synthetic_proteomicData</span>,</span></span>
<span class="r-in"><span>                       <span class="st">"Olink"</span> <span class="op">=</span> <span class="va">synthetic_OlinkData</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">deggs_object</span> <span class="op">&lt;-</span> <span class="fu">get_diffNetworks</span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">assayData_list</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span></span>
<span class="r-in"><span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span></span>
<span class="r-in"><span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span></span>
<span class="r-in"><span>                                 padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span></span>
<span class="r-in"><span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span>                                 </span></span>
<span class="r-in"><span><span class="co"># to use only certain categories for comparison: </span></span></span>
<span class="r-in"><span><span class="co"># let's randomly add another level of response to the example metadata</span></span></span>
<span class="r-in"><span><span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">indices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">synthetic_metadata</span><span class="op">)</span>, <span class="fl">20</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </span></span>
<span class="r-in"><span><span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span><span class="op">[</span><span class="va">indices</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Moderate response"</span></span></span>
<span class="r-in"><span><span class="va">deggs_object</span> <span class="op">&lt;-</span> <span class="fu">get_diffNetworks</span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">assayData_list</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span></span>
<span class="r-in"><span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span></span>
<span class="r-in"><span>                                 category_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Responder"</span>, </span></span>
<span class="r-in"><span>                                                     <span class="st">"Non_responder"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span></span>
<span class="r-in"><span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span>                                 </span></span>
<span class="r-in"><span><span class="co"># to be more generous on the targets to be excluded, and lower the expression </span></span></span>
<span class="r-in"><span><span class="co"># level threshold to the 25th percentile (or lower): </span></span></span>
<span class="r-in"><span><span class="va">deggs_object</span> <span class="op">&lt;-</span> <span class="fu">get_diffNetworks</span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">assayData_list</span>,</span></span>
<span class="r-in"><span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span></span>
<span class="r-in"><span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span></span>
<span class="r-in"><span>                                 category_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Responder"</span>, </span></span>
<span class="r-in"><span>                                                     <span class="st">"Non_responder"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span></span>
<span class="r-in"><span>                                 percentile_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.98</span>, by <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                 cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>                                  </span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Elisabetta Sciacca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

