<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-gb">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Differential Network Analysis with multiDEGGs • multiDEGGs</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential Network Analysis with multiDEGGs">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">multiDEGGs</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.2.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Differential_Network_Analisis.html">Differential Network Analysis with multiDEGGs</a></li>
    <li><a class="dropdown-item" href="../articles/Feature_Selection.html">Feature Selection and Feature Engineering with multiDEGGs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/elisabettasciacca/multiDEGGs/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Differential Network Analysis with multiDEGGs</h1>
                        <h4 data-toc-skip class="author">Elisabetta
Sciacca, Myles Lewis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/elisabettasciacca/multiDEGGs/blob/main/vignettes/Differential_Network_Analisis.Rmd" class="external-link"><code>vignettes/Differential_Network_Analisis.Rmd</code></a></small>
      <div class="d-none name"><code>Differential_Network_Analisis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The multiDEGGs package performs multi-omic differential network
analysis by identifying differential interactions between molecular
entities (genes, proteins, miRNAs, or other biomolecules) across the
omic datasets provided.</p>
<p>For each omic dataset, a differential network is constructed, where
links represent statistically significant differential interactions
between entities. These networks are then integrated into a
comprehensive visualization using distinct colors to distinguish
interactions from different omic layers. This unified visualization
allows interactive exploration of cross-omic patterns (e.g.,
differential interactions present at both transcript and protein level).
For each link, users can access differential statistical significance
metrics (p-values or adjusted p-values, calculated via robust or
traditional linear regression with interaction term), and differential
regression plots.</p>
<p>Beyond network visualization and exploration, multiDEGGs extends its
utility into predictive modeling applications. The identified
differential interactions can be leveraged as engineered features in
machine learning pipelines, providing biologically meaningful predictors
that capture relational information between molecular entities. The
package includes specialized functions for nested cross-validation that
ensure proper feature selection and engineering without data leakage,
enabling the construction of robust and interpretable predictive models
(see
<code>vignette("2. Feature Selection", package = "multiDEGGs")</code>
for details).</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install from CRAN:<br><code>install.packages("multiDEGGs")</code></p>
<p>Install from Github:<br><code>devtools::install_github("elisabettasciacca/multiDEGGs")</code></p>
</div>
<div class="section level2">
<h2 id="quick-start---generate-differential-networks">Quick start - Generate Differential Networks<a class="anchor" aria-label="anchor" href="#quick-start---generate-differential-networks"></a>
</h2>
<p>If you are working with human data, you can start differential
analysis by using the internal multiDEGGs’ default reference
network.<br>
Users working with other species or requiring custom interaction sets
can provide their own biological reference netowork as detailed in the
last paragraph of this vignette.</p>
<p>Let’s start by loading the package and sample data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/elisabettasciacca/multiDEGGs/" class="external-link">multiDEGGs</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_metadata"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_rnaseqData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_proteomicData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_OlinkData"</span><span class="op">)</span></span></code></pre></div>
<p>Generate Differential Networks:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assayData_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"RNAseq"</span> <span class="op">=</span> <span class="va">synthetic_rnaseqData</span>,</span>
<span>                       <span class="st">"Proteomics"</span> <span class="op">=</span> <span class="va">synthetic_proteomicData</span>,</span>
<span>                       <span class="st">"Olink"</span> <span class="op">=</span> <span class="va">synthetic_OlinkData</span><span class="op">)</span></span>
<span></span>
<span><span class="va">deggs_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_diffNetworks.html">get_diffNetworks</a></span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">assayData_list</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span>
<span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span>
<span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span>
<span>                                 padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="key-parameters-of-get_diffnetworks">Key Parameters of <code>get_diffNetworks</code><a class="anchor" aria-label="anchor" href="#key-parameters-of-get_diffnetworks"></a>
</h3>
<p>It’s worth explaining some of the important parameters of
<code>get_diffNetworks</code>:</p>
<ul>
<li><p><code>assayData</code>: accepts either a single normalized
matrix/data frame (for single omic differential analysis), or a list of
matrices/data frames (for multi-omic scenarios). For multi-omic
analysis, it’s highly recommended to use a named list of data. If
unnamed, sequential names (assayData1, assayData2, etc.) will be
assigned to identify each matrix or data frame.</p></li>
<li><p><code>metadata</code>: can also be a named factor vector, with
names matching the patient IDs in column names of the assay data
matrices/data frames. In that case, the category_variable can remain
unset (NULL by default).</p></li>
<li><p><code>category_subset</code>: this parameter can restrict the
analysis to a certain subset of categories available in the
metadata/category vector.</p></li>
<li><p><code>regression_method</code>: set to <code>"lm"</code> by
default because it is faster and highly recommended in machine learning
scenarios, where the function might be repeatedly called many times. For
basic differential analyses, <code>"rlm"</code> can also be used and may
perform better in some cases.</p></li>
<li><p><code>percentile_vector</code>: by default, molecular targets
(genes, proteins, etc.) whose expression level is below the 35th
percentile of the entire data matrix are excluded from the analysis.
This threshold can be modified by specifying the percentile vector that
is internally used for the percolation analysis. For example, to remove
only targets below the 25th percentile, set
<code>percentile_vector = seq(0.25, 0.98, by = 0.05)</code>.</p></li>
<li><p><code>padj_method</code>: the default method is Bonferroni.
Storey’s q values often give more generous results but the
<code>qvalue</code> package needs to be installed first.</p></li>
</ul>
<p><strong>NOTE</strong>: Not all patient IDs need to be present across
datasets. Different numbers of samples per omic are acceptable. Only IDs
whose data is available in the colnames of the assayData will be
included in the analysis. Missing IDs will be listed in a message
similar to:</p>
<p><code>The following samples IDs are missing in Proteomics: PT001, PT005, PT0030</code></p>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>The <code>deggs_object</code> now contains the differential networks
for each omic data in <code>assayData_list</code>. These networks can be
integrated into a comprehensive visualization where different colors
distinguish links from different omic layers.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/View_diffNetworks.html">View_diffNetworks</a></span><span class="op">(</span><span class="va">deggs_object</span><span class="op">)</span></span></code></pre></div>
<p>This visualization interface allows to:</p>
<ol style="list-style-type: decimal">
<li>Navigate the networks associated with each patient category</li>
<li>Filter by link significance</li>
<li>Search for specific genes inside the network</li>
</ol>
<p>
 
</p>
<img src="multiDEGGs_1.png" style="width:75.0%"><p>
 
</p>
<p>Thicker links correspond to higher significant p-values.<br>
The direction of the arrows shows the relationship direction reported in
literature, not derived from the data.</p>
<p>The user can visualize differential regression plots by clicking on a
link:</p>
<p>
 
</p>
<img src="multiDEGGs_2.png" style="width:75.0%"><p>
 
</p>
<p><br> This <strong>visual inspection of individual links is
essential</strong> before drawing any biological conclusions. These
regression plots allow assessment of:</p>
<ul>
<li>
<strong>Effect size</strong>: the magnitude of the difference in
interaction strength between groups</li>
<li>
<strong>Model fit quality</strong>: how well the regression lines
represent the data</li>
<li>
<strong>Noise levels</strong>: the scatter and confidence intervals
around the regression lines</li>
<li>
<strong>Outliers</strong>: presence of influential data points that
might drive the observed difference</li>
</ul>
<p><br> Single node differential expressions can also be visualized by
clicking on the nodes:</p>
<p>
 
</p>
<img src="multiDEGGs_3.png" style="width:80.0%"><p>
 
</p>
<p><strong>NOTE</strong>: For multi-omic scenarios, the data from the
first matrix in the list passed to <code>assayData</code> will be used
for this boxplot.</p>
</div>
<div class="section level2">
<h2 id="list-all-differential-interactions">List All Differential Interactions<a class="anchor" aria-label="anchor" href="#list-all-differential-interactions"></a>
</h2>
<p>Outside of the interactive environment, the
<code><a href="../reference/get_multiOmics_diffNetworks.html">get_multiOmics_diffNetworks()</a></code> function can be used to get a
table of all differential interactions, ordered by p-value or adjusted
p-value:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_multiOmics_diffNetworks.html">get_multiOmics_diffNetworks</a></span><span class="op">(</span><span class="va">deggs_object</span>, sig_threshold <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Non_responder</span></span>
<span><span class="co">#&gt;                  from       to      p.value        p.adj      layer</span></span>
<span><span class="co">#&gt; TNF-TNFRSF1A      TNF TNFRSF1A 3.325023e-04 1.330009e-03     RNAseq</span></span>
<span><span class="co">#&gt; IL1B-IL1R2       IL1B    IL1R2 5.389195e-04 2.155678e-03     RNAseq</span></span>
<span><span class="co">#&gt; TGFB3-TGFBR1    TGFB3   TGFBR1 5.329071e-15 2.131628e-14     RNAseq</span></span>
<span><span class="co">#&gt; AKT2-MTOR        AKT2     MTOR 4.347083e-04 1.738833e-03     RNAseq</span></span>
<span><span class="co">#&gt; FANCD2-FAN1    FANCD2     FAN1 4.440892e-16 3.552714e-15 Proteomics</span></span>
<span><span class="co">#&gt; GNG12-RASA2     GNG12    RASA2 0.000000e+00 0.000000e+00 Proteomics</span></span>
<span><span class="co">#&gt; RASGRP3-RRAS  RASGRP3     RRAS 2.202344e-04 1.761875e-03 Proteomics</span></span>
<span><span class="co">#&gt; TNF-TNFRSF1A1     TNF TNFRSF1A 0.000000e+00 0.000000e+00 Proteomics</span></span>
<span><span class="co">#&gt; RASGRP1-RAP1A RASGRP1    RAP1A 2.840564e-03 2.272452e-02 Proteomics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Responder</span></span>
<span><span class="co">#&gt;                from    to      p.value        p.adj  layer</span></span>
<span><span class="co">#&gt; FANCD2-FAN1  FANCD2  FAN1 0.000000e+00 0.000000e+00 RNAseq</span></span>
<span><span class="co">#&gt; FASLG-FAS     FASLG   FAS 6.947839e-07 2.084352e-06 RNAseq</span></span>
<span><span class="co">#&gt; MAP2K2-MAPK3 MAP2K2 MAPK3 1.153300e-12 3.459899e-12 RNAseq</span></span></code></pre></div>
<p>
For single omic scenarios, use the <code><a href="../reference/get_sig_deggs.html">get_sig_deggs()</a></code>
function:
</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deggs_object_oneOmic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_diffNetworks.html">get_diffNetworks</a></span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">synthetic_rnaseqData</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span>
<span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span>
<span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span>
<span>                                 padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/get_sig_deggs.html">get_sig_deggs</a></span><span class="op">(</span><span class="va">deggs_object_oneOmic</span>, sig_threshold <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt;                              from       to      p.value        p.adj</span></span>
<span><span class="co">#&gt; Non_responder.TNF-TNFRSF1A    TNF TNFRSF1A 3.325023e-04 1.330009e-03</span></span>
<span><span class="co">#&gt; Non_responder.IL1B-IL1R2     IL1B    IL1R2 5.389195e-04 2.155678e-03</span></span>
<span><span class="co">#&gt; Non_responder.TGFB3-TGFBR1  TGFB3   TGFBR1 5.329071e-15 2.131628e-14</span></span>
<span><span class="co">#&gt; Non_responder.AKT2-MTOR      AKT2     MTOR 4.347083e-04 1.738833e-03</span></span>
<span><span class="co">#&gt; Responder.FANCD2-FAN1      FANCD2     FAN1 0.000000e+00 0.000000e+00</span></span>
<span><span class="co">#&gt; Responder.FASLG-FAS         FASLG      FAS 6.947839e-07 2.084352e-06</span></span>
<span><span class="co">#&gt; Responder.MAP2K2-MAPK3     MAP2K2    MAPK3 1.153300e-12 3.459899e-12</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-regression-plots">Differential Regression Plots<a class="anchor" aria-label="anchor" href="#differential-regression-plots"></a>
</h2>
<p>To plot the differential regression fits outside of the interactive
environment, use <code><a href="../reference/plot_regressions.html">plot_regressions()</a></code> specifying the omic
data to be used and the two targets:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_regressions.html">plot_regressions</a></span><span class="op">(</span><span class="va">deggs_object</span>,</span>
<span>                 assayDataName <span class="op">=</span> <span class="st">"RNAseq"</span>,</span>
<span>                 gene_A <span class="op">=</span> <span class="st">"MTOR"</span>, </span>
<span>                 gene_B <span class="op">=</span> <span class="st">"AKT2"</span>,</span>
<span>                 legend_position <span class="op">=</span> <span class="st">"bottomright"</span><span class="op">)</span></span></code></pre></div>
<p>
 
</p>
<p><img src="plot_regressions.png" style="width:50.0%"></p>
<p>In single omic analyses, the <code>assayDataName</code> parameter can
remain unset.</p>
</div>
<div class="section level2">
<h2 id="differential-network-analysis-with-more-than-two-groups">Differential Network Analysis with More Than Two Groups<a class="anchor" aria-label="anchor" href="#differential-network-analysis-with-more-than-two-groups"></a>
</h2>
<p>It’s possible to compare differential interactions among more than
two categorical groups. All steps described above stay the same;<br>
the dropdown menu of the interactive environment will show all available
categories:</p>
<table class="table"><tr>
<td>
<img src="multiDEGGs_4.png" width="70%">
</td>
</tr></table>
<p>
</p>
<p>
While regressions and boxplots will show all categories:
</p>
<table class="table"><tr>
<td>
<img src="multiDEGGs_5.png" width="100%">
</td>
<td>
<img src="multiDEGGs_6.png" width="100%">
</td>
</tr></table>
<p>The statistical significance of the interaction term is calculated
via one-way ANOVA in this case.<br>
We highly recommend to have at least 4 or 5 observations per group.</p>
</div>
<div class="section level2">
<h2 id="custom-reference-network">Custom Reference Network<a class="anchor" aria-label="anchor" href="#custom-reference-network"></a>
</h2>
<p>The multiDEGGs package uses a human-specific reference network by
default, consisting of 10,537 molecular interactions (from KEGG,
mirTARbase, miRecords, and transmiR) obtained via the
<code>exportgraph</code> function of the MITHrIL tool (Alaimo et al.,
Oncotarget, 2016).</p>
<p>Users working with non-human species or requiring custom interaction
sets can provide their own reference network using the
<code>network</code> parameter of the <code><a href="../reference/get_diffNetworks.html">get_diffNetworks()</a></code>
function.<br><br></p>
<p><strong>Network Requirements</strong><br>
The custom network must be a <code>data.frame</code> with exactly two
columns:</p>
<ul>
<li>
<code>from</code>: Origin node (e.g., Transcription Factor, miRNA,
or Gene)</li>
<li>
<code>to</code>: Target node</li>
</ul>
<p>Nodes should use the same nomenclature (e.g., Gene Symbols, Entrez
IDs) as the expression assay data to ensure proper mapping during the
differential network analysis.<br><br></p>
<p><strong>Sources for Other Species</strong><br>
If you are working with non-human species, several repositories allow
for the extraction of interaction data:</p>
<ul>
<li>
<strong>STRING-db</strong>: Ideal for protein-protein interactions
(PPI) across thousands of species. You can download the “protein.links”
files</li>
<li>
<strong>KEGG</strong> (via <code>KEGGgraph</code> or
<code>TIGER</code>): Useful for metabolic and signaling pathways</li>
<li>
<strong>Omnipath</strong>: A comprehensive resource that integrates
multiple databases for human and mouse, easily accessible via the
<code>OmnipathR</code> package</li>
<li>
<strong>miRBase</strong> or <strong>TargetScan</strong>: For those
needing specific miRNA-target interactions for non-human models</li>
</ul>
<p><br></p>
<p><strong>Example: Generating a Custom Network in R</strong><br>
Below is a practical example of how to prepare a custom network,
assuming you have a list of interactions:</p>
<pre><code><span><span class="co"># Example: Creating a toy network for a non-human species</span></span>
<span><span class="co"># In a real scenario, you might load a CSV or use a database-specific package</span></span>
<span><span class="va">interaction_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  source <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GeneA"</span>, <span class="st">"GeneB"</span>, <span class="st">"miRNA1"</span><span class="op">)</span>,</span>
<span>  target <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GeneC"</span>, <span class="st">"GeneA"</span>, <span class="st">"GeneB"</span><span class="op">)</span>,</span>
<span>  confidence <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">0.8</span>, <span class="fl">0.95</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Formatting for get_diffNetworks() </span></span>
<span><span class="co"># We only need the 'from' and 'to' columns</span></span>
<span><span class="va">custom_network</span> <span class="op">&lt;-</span> <span class="va">interaction_data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"source"</span>, <span class="st">"target"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">custom_network</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"from"</span>, <span class="st">"to"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check class and structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">custom_network</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">custom_network</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Now this can be passed to the function:</span></span>
<span><span class="va">diff_net</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_diffNetworks.html">get_diffNetworks</a></span><span class="op">(</span><span class="va">expr_data</span>, network <span class="op">=</span> <span class="va">custom_network</span>, <span class="va">...</span><span class="op">)</span></span></code></pre>
<p><br></p>
<p><strong>NOTE</strong><br>
Finally, it is important point to keep in mind that the differential
networks obtained will be fundamentally limited by the reference
biological network used. multiDEGGs can only identify interactions that
are already present in the reference network, so using an incomplete or
biased network might cause missing relevant biological relationships or,
conversely, finding spurious associations. When choosing a reference
network, it is important to assess its source, how comprehensive it is,
and the quality of evidence supporting the interactions. These aspects
should be documented when reporting multiDEGGs results.</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"multiDEGGs"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'multiDEGGs' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Sciacca E, et al. (2023). "DEGGs: An R package with shiny app for the</span></span>
<span><span class="co">#&gt;   identification of differentially expressed gene-gene interactions in</span></span>
<span><span class="co">#&gt;   high-Throughput sequencing data." _Bioinformatics_, *39*, btad192.</span></span>
<span><span class="co">#&gt;   doi:10.1093/bioinformatics/btad192</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1093/bioinformatics/btad192&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {DEGGs: An R package with shiny app for the identification of differentially expressed gene-gene interactions in high-Throughput sequencing data},</span></span>
<span><span class="co">#&gt;     author = {Elisabetta Sciacca and {et al.}},</span></span>
<span><span class="co">#&gt;     journal = {Bioinformatics},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     volume = {39},</span></span>
<span><span class="co">#&gt;     pages = {btad192},</span></span>
<span><span class="co">#&gt;     doi = {10.1093/bioinformatics/btad192},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Elisabetta Sciacca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
