<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-gb">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Differential Network Analysis with multiDEGGs • multiDEGGs</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential Network Analysis with multiDEGGs">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">multiDEGGs</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/multiDEGGs_vignette.html">Differential Network Analysis with multiDEGGs</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/elisabettasciacca/multiDEGGs/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Differential Network Analysis with multiDEGGs</h1>
                        <h4 data-toc-skip class="author">Elisabetta
Sciacca, Myles Lewis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/elisabettasciacca/multiDEGGs/blob/v1.1.0/vignettes/multiDEGGs_vignette.Rmd" class="external-link"><code>vignettes/multiDEGGs_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>multiDEGGs_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The multiDEGGs package performs multi-omic differential network
analysis by identifying differential interactions between molecular
entities (genes, proteins, miRNAs, or other biomolecules) across the
omic datasets provided.</p>
<p>For each omic dataset, a differential network is constructed, where
links represent statistically significant differential interactions
between entities. These networks are then integrated into a
comprehensive visualization using distinct colors to distinguish
interactions from different omic layers. This unified visualization
allows interactive exploration of cross-omic patterns (e.g.,
differential interactions present at both transcript and protein level).
For each link, users can access differential statistical significance
metrics (p-values or adjusted p-values, calculated via robust or
traditional linear regression with interaction term), and differential
regression plots.</p>
<p>Beyond network visualization and exploration, multiDEGGs extends its
utility into predictive modeling applications. The identified
differential interactions can be leveraged as engineered features in
machine learning pipelines, providing biologically meaningful predictors
that capture relational information between molecular entities. The
package includes specialized functions for nested cross-validation that
ensure proper feature selection and engineering without data leakage,
enabling the construction of robust and interpretable predictive
models.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install from CRAN:<br><code>install.packages("multiDEGGs")</code></p>
<p>Install from Github:<br><code>devtools::install_github("elisabettasciacca/multiDEGGs")</code></p>
</div>
<div class="section level2">
<h2 id="quick-start---generate-differential-networks">Quick start - Generate Differential Networks<a class="anchor" aria-label="anchor" href="#quick-start---generate-differential-networks"></a>
</h2>
<p>Let’s start by loading the package and sample data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/elisabettasciacca/multiDEGGs/" class="external-link">multiDEGGs</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_metadata"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_rnaseqData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_proteomicData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_OlinkData"</span><span class="op">)</span></span></code></pre></div>
<p>Generate Differential Networks:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assayData_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"RNAseq"</span> <span class="op">=</span> <span class="va">synthetic_rnaseqData</span>,</span>
<span>                       <span class="st">"Proteomics"</span> <span class="op">=</span> <span class="va">synthetic_proteomicData</span>,</span>
<span>                       <span class="st">"Olink"</span> <span class="op">=</span> <span class="va">synthetic_OlinkData</span><span class="op">)</span></span>
<span></span>
<span><span class="va">deggs_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_diffNetworks.html">get_diffNetworks</a></span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">assayData_list</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span>
<span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span>
<span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span>
<span>                                 padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="key-parameters-of-get_diffnetworks">Key Parameters of <code>get_diffNetworks</code><a class="anchor" aria-label="anchor" href="#key-parameters-of-get_diffnetworks"></a>
</h3>
<p>It’s worth explaining some of the important parameters of
<code>get_diffNetworks</code>:</p>
<ul>
<li><p><code>assayData</code>: accepts either a single normalized
matrix/data frame (for single omic differential analysis), or a list of
matrices/data frames (for multi-omic scenarios). For multi-omic
analysis, it’s highly recommended to use a named list of data. If
unnamed, sequential names (assayData1, assayData2, etc.) will be
assigned to identify each matrix or data frame.</p></li>
<li><p><code>metadata</code>: can also be a named factor vector, with
names matching the patient IDs in column names of the assay data
matrices/data frames. In that case, the category_variable can remain
unset (NULL by default).</p></li>
<li><p><code>category_subset</code>: this parameter can restrict the
analysis to a certain subset of categories available in the
metadata/category vector.</p></li>
<li><p><code>regression_method</code>: set to <code>"lm"</code> by
default because it is faster and highly recommended in machine learning
scenarios, where the function might be repeatedly called many times. For
basic differential analyses, <code>"rlm"</code> can also be used and may
perform better in some cases.</p></li>
<li><p><code>percentile_vector</code>: by default, molecular targets
(genes, proteins, etc.) whose expression level is below the 35th
percentile of the entire data matrix are excluded from the analysis.
This threshold can be modified by specifying the percentile vector that
is internally used for the percolation analysis. For example, to remove
only targets below the 25th percentile, set
<code>percentile_vector = seq(0.25, 0.98, by = 0.05)</code>.</p></li>
<li><p><code>padj_method</code>: the default method is Bonferroni.
Storey’s q values often give more generous results but the
<code>qvalue</code> package needs to be installed first.</p></li>
</ul>
<p><strong>NOTE</strong>: Not all patient IDs need to be present across
datasets. Different numbers of samples per omic are acceptable. Only IDs
whose data is available in the colnames of the assayData will be
included in the analysis. Missing IDs will be listed in a message
similar to:</p>
<p><code>The following samples IDs are missing in Proteomics: PT001, PT005, PT0030</code></p>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>The <code>deggs_object</code> now contains the differential networks
for each omic data in <code>assayData_list</code>. These networks can be
integrated into a comprehensive visualization where different colors
distinguish links from different omic layers.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/View_diffNetworks.html">View_diffNetworks</a></span><span class="op">(</span><span class="va">deggs_object</span><span class="op">)</span></span></code></pre></div>
<p>This visualization interface allows you to:</p>
<ol style="list-style-type: decimal">
<li>Navigate the networks associated with each patient category</li>
<li>Filter by link significance</li>
<li>Search for specific genes inside the network</li>
</ol>
<p>
 
</p>
<img src="multiDEGGs_1.png" style="width:75.0%"><p>
 
</p>
<p>Thicker links correspond to higher significant p-values.<br>
The direction of the arrows shows the relationship direction reported in
literature, not derived from the data.</p>
<p>The user can visualize differential regression plots by clicking on a
link:</p>
<p>
 
</p>
<img src="multiDEGGs_2.png" style="width:75.0%"><p>
 
</p>
<p>Single node differential expressions can also be visualized by
clicking on the nodes:</p>
<p>
 
</p>
<img src="multiDEGGs_3.png" style="width:80.0%"><p>
 
</p>
<p><strong>NOTE</strong>: For multi-omic scenarios, the data from the
first matrix in the list passed to <code>assayData</code> will be used
for this boxplot.</p>
</div>
<div class="section level2">
<h2 id="list-all-differential-interactions">List All Differential Interactions<a class="anchor" aria-label="anchor" href="#list-all-differential-interactions"></a>
</h2>
<p>Outside of the interactive environment, the
<code><a href="../reference/get_multiOmics_diffNetworks.html">get_multiOmics_diffNetworks()</a></code> function can be used to get a
table of all differential interactions, ordered by p-value or adjusted
p-value:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_multiOmics_diffNetworks.html">get_multiOmics_diffNetworks</a></span><span class="op">(</span><span class="va">deggs_object</span>, sig_threshold <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; $Non_responder</span></span>
<span><span class="co">#&gt;                  from       to      p.value        p.adj      layer</span></span>
<span><span class="co">#&gt; TNF-TNFRSF1A      TNF TNFRSF1A 3.325023e-04 1.330009e-03     RNAseq</span></span>
<span><span class="co">#&gt; IL1B-IL1R2       IL1B    IL1R2 5.389195e-04 2.155678e-03     RNAseq</span></span>
<span><span class="co">#&gt; TGFB3-TGFBR1    TGFB3   TGFBR1 5.329071e-15 2.131628e-14     RNAseq</span></span>
<span><span class="co">#&gt; AKT2-MTOR        AKT2     MTOR 4.347083e-04 1.738833e-03     RNAseq</span></span>
<span><span class="co">#&gt; FANCD2-FAN1    FANCD2     FAN1 4.440892e-16 3.552714e-15 Proteomics</span></span>
<span><span class="co">#&gt; GNG12-RASA2     GNG12    RASA2 0.000000e+00 0.000000e+00 Proteomics</span></span>
<span><span class="co">#&gt; RASGRP3-RRAS  RASGRP3     RRAS 2.202344e-04 1.761875e-03 Proteomics</span></span>
<span><span class="co">#&gt; TNF-TNFRSF1A1     TNF TNFRSF1A 0.000000e+00 0.000000e+00 Proteomics</span></span>
<span><span class="co">#&gt; RASGRP1-RAP1A RASGRP1    RAP1A 2.840564e-03 2.272452e-02 Proteomics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Responder</span></span>
<span><span class="co">#&gt;                from    to      p.value        p.adj  layer</span></span>
<span><span class="co">#&gt; FANCD2-FAN1  FANCD2  FAN1 0.000000e+00 0.000000e+00 RNAseq</span></span>
<span><span class="co">#&gt; FASLG-FAS     FASLG   FAS 6.947839e-07 2.084352e-06 RNAseq</span></span>
<span><span class="co">#&gt; MAP2K2-MAPK3 MAP2K2 MAPK3 1.153300e-12 3.459899e-12 RNAseq</span></span></code></pre></div>
<p>
For single omic scenarios, use the <code><a href="../reference/get_sig_deggs.html">get_sig_deggs()</a></code>
function:
</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deggs_object_oneOmic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_diffNetworks.html">get_diffNetworks</a></span><span class="op">(</span>assayData <span class="op">=</span> <span class="va">synthetic_rnaseqData</span>,</span>
<span>                                 metadata <span class="op">=</span> <span class="va">synthetic_metadata</span>,</span>
<span>                                 category_variable <span class="op">=</span> <span class="st">"response"</span>,</span>
<span>                                 regression_method <span class="op">=</span> <span class="st">"lm"</span>,</span>
<span>                                 padj_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>                                 verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 show_progressBar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                 cores <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/get_sig_deggs.html">get_sig_deggs</a></span><span class="op">(</span><span class="va">deggs_object_oneOmic</span>, sig_threshold <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt;                              from       to      p.value        p.adj</span></span>
<span><span class="co">#&gt; Non_responder.TNF-TNFRSF1A    TNF TNFRSF1A 3.325023e-04 1.330009e-03</span></span>
<span><span class="co">#&gt; Non_responder.IL1B-IL1R2     IL1B    IL1R2 5.389195e-04 2.155678e-03</span></span>
<span><span class="co">#&gt; Non_responder.TGFB3-TGFBR1  TGFB3   TGFBR1 5.329071e-15 2.131628e-14</span></span>
<span><span class="co">#&gt; Non_responder.AKT2-MTOR      AKT2     MTOR 4.347083e-04 1.738833e-03</span></span>
<span><span class="co">#&gt; Responder.FANCD2-FAN1      FANCD2     FAN1 0.000000e+00 0.000000e+00</span></span>
<span><span class="co">#&gt; Responder.FASLG-FAS         FASLG      FAS 6.947839e-07 2.084352e-06</span></span>
<span><span class="co">#&gt; Responder.MAP2K2-MAPK3     MAP2K2    MAPK3 1.153300e-12 3.459899e-12</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-regression-plots">Differential Regression Plots<a class="anchor" aria-label="anchor" href="#differential-regression-plots"></a>
</h2>
<p>To plot the differential regression fits outside of the interactive
environment, use <code><a href="../reference/plot_regressions.html">plot_regressions()</a></code> specifying the omic
data to be used and the two targets:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_regressions.html">plot_regressions</a></span><span class="op">(</span><span class="va">deggs_object</span>,</span>
<span>                 assayDataName <span class="op">=</span> <span class="st">"RNAseq"</span>,</span>
<span>                 gene_A <span class="op">=</span> <span class="st">"MTOR"</span>, </span>
<span>                 gene_B <span class="op">=</span> <span class="st">"AKT2"</span>,</span>
<span>                 legend_position <span class="op">=</span> <span class="st">"bottomright"</span><span class="op">)</span></span></code></pre></div>
<p>
 
</p>
<p><img src="plot_regressions.png" style="width:50.0%"></p>
<p>In single omic analyses, the <code>assayDataName</code> parameter can
remain unset.</p>
</div>
<div class="section level2">
<h2 id="differential-network-analysis-with-more-than-two-groups">Differential Network Analysis with More Than Two Groups<a class="anchor" aria-label="anchor" href="#differential-network-analysis-with-more-than-two-groups"></a>
</h2>
<p>It’s possible to compare differential interactions among more than
two categorical groups. All steps described above stay the same;<br>
the dropdown menu of the interactive environment will show all available
categories:</p>
<table class="table"><tr>
<td>
<img src="multiDEGGs_4.png" width="70%">
</td>
</tr></table>
<p>
</p>
<p>
While regressions and boxplots will show all categories:
</p>
<table class="table"><tr>
<td>
<img src="multiDEGGs_5.png" width="100%">
</td>
<td>
<img src="multiDEGGs_6.png" width="100%">
</td>
</tr></table>
<p>The statistical significance of the interaction term is calculated
via one-way ANOVA in this case.<br>
We highly recommend to have at least 4 or 5 observations per group.</p>
</div>
<div class="section level2">
<h2 id="feature-selection-and-engineering-with-multideggs-in-nested-cross-validation">Feature Selection and Engineering with multiDEGGs in Nested
Cross-Validation<a class="anchor" aria-label="anchor" href="#feature-selection-and-engineering-with-multideggs-in-nested-cross-validation"></a>
</h2>
<p>In computational biology applications involving high-throughput data,
researchers commonly encounter situations where the number of potential
predictors far exceeds the available sample size. This dimensional
challenge requires careful feature selection strategies for both
mathematical and clinical reasons.</p>
<p>Standard feature selection methods typically evaluate predictors
individually, identifying those variables that show the strongest
univariate associations with the outcome variable (such as through
t-tests or Wilcoxon tests). While effective, this approach overlooks the
interconnected nature of biological systems, where </p>
<p>Feature engineering represents a complementary strategy that creates
new predictors by combining or transforming existing variables. In
biology, such approach can be used to capture higher-order information
that reflects the interconnected nature of molecular processes. For
instance, the ratio between two genes may provide more discriminative
power than either gene expression level independently, particularly when
their relative balance is disrupted in disease states.</p>
<p>The informative content encoded in differential interactions,
combined with multiDEGGs’ ability to identify only literature-validated
differential relationships, makes it particularly well-suited for both
individual feature selection and guided creation of engineered
predictors in machine learning. Such approach has potential to overcome
the limitations of conventional algorithms which may select individual
predictors without clear biological significance, compromising both the
interpretability and clinical credibility of the resulting models.</p>
<div class="section level3">
<h3 id="why-nested-cross-validation-for-feature-engineering">Why Nested Cross-Validation for Feature Engineering?<a class="anchor" aria-label="anchor" href="#why-nested-cross-validation-for-feature-engineering"></a>
</h3>
<p>It is crucial that feature selection and modification is conducted
exclusively on training data within cross-validation loops to prevent
information leakage from the test set. The <code>nestedcv</code> package
enables the nested modification of predictors within each outer fold,
ensuring that the attributes learned from the training part are applied
to the test data without prior knowledge of the test data itself.<br>
The selected and combined features, and corresponding model, can then be
evaluated on the hold-out test data without introducing bias.</p>
<p>Both (nestcv.glmnet) and (nestcv.train) from <code>nestedcv</code>
accept any user-defined function that filters or transforms the feature
matrix by passing the function name to the <code>modifyX</code>
parameter.<br><strong>The multiDEGGs package provides two specialized functions for
this purpose.</strong></p>
</div>
<div class="section level3">
<h3 id="multideggs_filter-pure-differential-network-based-selection">multiDEGGs_filter(): Pure Differential Network-Based Selection<a class="anchor" aria-label="anchor" href="#multideggs_filter-pure-differential-network-based-selection"></a>
</h3>
<p>The <code><a href="../reference/multiDEGGs_filter.html">multiDEGGs_filter()</a></code> function performs feature
selection based entirely on differential network analysis. It identifies
significant differential molecular interactions and can return either
the interaction pairs alone or both pairs and individual variables
involved in those interactions.</p>
<div class="section level4">
<h4 id="key-parameters">Key Parameters<a class="anchor" aria-label="anchor" href="#key-parameters"></a>
</h4>
<p>When using <code><a href="../reference/multiDEGGs_filter.html">multiDEGGs_filter()</a></code>, you can control the
following parameters through <code>modifyX_options</code>:</p>
<ul>
<li>
<strong><code>keep_single_genes</code></strong> (logical, default
<code>FALSE</code>): Controls whether to include individual genes from
significant pairs in addition to the pairs themselves</li>
<li>
<strong><code>nfilter</code></strong> (integer, default
<code>NULL</code>): Maximum number of predictors to return. When
<code>NULL</code>, all significant interactions found are included</li>
</ul>
</div>
<div class="section level4">
<h4 id="usage-examples">Usage Examples<a class="anchor" aria-label="anchor" href="#usage-examples"></a>
</h4>
<div class="section level5">
<h5 id="basic-usage-pairs-only">Basic Usage: Pairs Only<a class="anchor" aria-label="anchor" href="#basic-usage-pairs-only"></a>
</h5>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/myles-lewis/nestedcv" class="external-link">nestedcv</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_metadata"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"synthetic_rnaseqData"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Regularized linear model with interaction pairs only</span></span>
<span><span class="va">fit.glmnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/nestedcv/man/nestcv.glmnet.html" class="external-link">nestcv.glmnet</a></span><span class="op">(</span></span>
<span>  y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span><span class="op">)</span>,</span>
<span>  x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">synthetic_rnaseqData</span><span class="op">)</span>,</span>
<span>  modifyX <span class="op">=</span> <span class="st">"multiDEGGs_filter"</span>,</span>
<span>  modifyX_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    keep_single_genes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    nfilter <span class="op">=</span> <span class="fl">20</span></span>
<span>  <span class="op">)</span>,</span>
<span>  modifyX_useY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_outer_folds <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  n_inner_folds <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">fit.glmnet</span><span class="op">)</span></span>
<span><span class="co">#&gt; Nested cross-validation with glmnet</span></span>
<span><span class="co">#&gt; No filter</span></span>
<span><span class="co">#&gt; Modifier:  multiDEGGs_filter </span></span>
<span><span class="co">#&gt; Outer loop:  5-fold CV</span></span>
<span><span class="co">#&gt; Inner loop:  6-fold CV</span></span>
<span><span class="co">#&gt; 100 observations, 14 predictors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;        alpha  lambda n.filter</span></span>
<span><span class="co">#&gt; Fold 1   0.2 0.16011        7</span></span>
<span><span class="co">#&gt; Fold 2   0.1 0.07423        7</span></span>
<span><span class="co">#&gt; Fold 3   0.1 0.14076        7</span></span>
<span><span class="co">#&gt; Fold 4   0.1 0.13511        7</span></span>
<span><span class="co">#&gt; Fold 5   1.0 0.15091        7</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Final parameters:</span></span>
<span><span class="co">#&gt;  lambda    alpha  </span></span>
<span><span class="co">#&gt; 0.07099  0.10000  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Final coefficients:</span></span>
<span><span class="co">#&gt;  (Intercept) TNF:TNFRSF1A    AKT2:MTOR   IL1B:IL1R2    FASLG:FAS TGFB3:TGFBR1 </span></span>
<span><span class="co">#&gt;     1.808953    -0.188626    -0.113628     0.050012    -0.034766    -0.030783 </span></span>
<span><span class="co">#&gt; MAP2K2:MAPK3  FANCD2:FAN1 </span></span>
<span><span class="co">#&gt;    -0.020656    -0.008568 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Result:</span></span>
<span><span class="co">#&gt;        RMSE     R.squared   Pearson.r^2           MAE   </span></span>
<span><span class="co">#&gt;     0.48505       0.03419       0.04929       0.46076</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="including-individual-genes-keep_single_genes-true">Including Individual Genes (keep_single_genes = TRUE)<a class="anchor" aria-label="anchor" href="#including-individual-genes-keep_single_genes-true"></a>
</h5>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Random forest model including both pairs and individual genes</span></span>
<span><span class="va">fit.rf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/nestedcv/man/nestcv.train.html" class="external-link">nestcv.train</a></span><span class="op">(</span></span>
<span>  y <span class="op">=</span> <span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span>,</span>
<span>  x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">synthetic_rnaseqData</span><span class="op">)</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"rf"</span>,</span>
<span>  modifyX <span class="op">=</span> <span class="st">"multiDEGGs_filter"</span>,</span>
<span>  modifyX_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    keep_single_genes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    nfilter <span class="op">=</span> <span class="fl">30</span></span>
<span>  <span class="op">)</span>,</span>
<span>  modifyX_useY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_outer_folds <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  n_inner_folds <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: ggplot2</span></span>
<span><span class="co">#&gt; Loading required package: lattice</span></span>
<span></span>
<span><span class="va">fit.rf</span><span class="op">$</span><span class="va">summary</span></span>
<span><span class="co">#&gt;                Reference</span></span>
<span><span class="co">#&gt; Predicted       Non_responder Responder</span></span>
<span><span class="co">#&gt;   Non_responder            57         1</span></span>
<span><span class="co">#&gt;   Responder                 1        41</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;               AUC            Accuracy   Balanced accuracy   </span></span>
<span><span class="co">#&gt;            0.9988              0.9800              0.9795</span></span>
<span></span>
<span><span class="co"># Plot ROC on outer folds</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fit.rf</span><span class="op">$</span><span class="va">roc</span><span class="op">)</span></span></code></pre></div>
<p><img src="multiDEGGs_vignette_files/figure-html/unnamed-chunk-8-1.png" width="288"></p>
</div>
</div>
<div class="section level4">
<h4 id="how-nfilter-works-with-keep_single_genes">How nfilter works with keep_single_genes<a class="anchor" aria-label="anchor" href="#how-nfilter-works-with-keep_single_genes"></a>
</h4>
<ul>
<li>When <strong><code>keep_single_genes = FALSE</code></strong>:
<code>nfilter</code> limits only the number of interaction pairs
returned</li>
<li>When <strong><code>keep_single_genes = TRUE</code></strong>:
<code>nfilter</code> limits the combined count of unique individual
genes plus interaction pairs. The function prioritizes pairs by
significance and adds individual genes as needed until the limit is
reached</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="multideggs_combined_filter-hybrid-statistical-and-network-based-selection">multiDEGGs_combined_filter(): Hybrid Statistical and Network-Based
Selection<a class="anchor" aria-label="anchor" href="#multideggs_combined_filter-hybrid-statistical-and-network-based-selection"></a>
</h3>
<p>The <code><a href="../reference/multiDEGGs_combined_filter.html">multiDEGGs_combined_filter()</a></code> function combines
traditional statistical feature selection with differential network
analysis. This hybrid approach allows you to benefit from both
conventional univariate selection methods and the biological insights
from interaction analysis.</p>
<div class="section level4">
<h4 id="key-parameters-1">Key Parameters<a class="anchor" aria-label="anchor" href="#key-parameters-1"></a>
</h4>
<ul>
<li>
<strong><code>filter_method</code></strong> (character): Statistical
method for single feature selection.<br>
Options: <code>"ttest"</code>, <code>"wilcoxon"</code>,
<code>"ranger"</code>, <code>"glmnet"</code>, <code>"pls"</code>
</li>
<li>
<strong><code>nfilter</code></strong> (integer): Maximum number of
features to select</li>
<li>
<strong><code>dynamic_nfilter</code></strong> (logical): Controls
how <code>nfilter</code> is applied (see detailed explanation
below)</li>
<li>
<strong><code>keep_single_genes</code></strong> (logical): When
<code>dynamic_nfilter = TRUE</code>, determines whether to include
individual genes from multiDEGGs pairs</li>
</ul>
</div>
<div class="section level4">
<h4 id="dynamic-vs--balanced-selection-modes">Dynamic vs. Balanced Selection Modes<a class="anchor" aria-label="anchor" href="#dynamic-vs--balanced-selection-modes"></a>
</h4>
<div class="section level5">
<h5 id="dynamic-selection-dynamic_nfilter-true">Dynamic Selection (<code>dynamic_nfilter = TRUE</code>)<a class="anchor" aria-label="anchor" href="#dynamic-selection-dynamic_nfilter-true"></a>
</h5>
<p>In dynamic mode, the function: 1. Selects <code>nfilter</code> single
genes using the chosen statistical method 2. Adds ALL significant
interaction pairs found by multiDEGGs 3. Total predictors =
<code>nfilter</code> single genes + number of significant pairs</p>
<p>This mode allows the feature space to expand based on the biological
complexity discovered in each fold.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dynamic selection with t-test for single genes</span></span>
<span><span class="va">fit.dynamic</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/nestedcv/man/nestcv.glmnet.html" class="external-link">nestcv.glmnet</a></span><span class="op">(</span></span>
<span>  y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span><span class="op">)</span>,</span>
<span>  x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">synthetic_rnaseqData</span><span class="op">)</span>,</span>
<span>  modifyX <span class="op">=</span> <span class="st">"multiDEGGs_combined_filter"</span>,</span>
<span>  modifyX_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    filter_method <span class="op">=</span> <span class="st">"ttest"</span>, </span>
<span>    nfilter <span class="op">=</span> <span class="fl">20</span>,</span>
<span>    dynamic_nfilter <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>    keep_single_genes <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span>,</span>
<span>  modifyX_useY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_outer_folds <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  n_inner_folds <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="balanced-selection-dynamic_nfilter-false">Balanced Selection (<code>dynamic_nfilter = FALSE</code>)<a class="anchor" aria-label="anchor" href="#balanced-selection-dynamic_nfilter-false"></a>
</h5>
<p>In balanced mode, the function:<br>
1. Allocates approximately half of <code>nfilter</code> to interaction
pairs<br>
2. Fills remaining slots with single genes from the statistical
filter<br>
3. Maintains consistent total number of predictors across all folds</p>
<p>This mode ensures a fixed feature space size while balancing single
genes and interactions.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Balanced selection with Wilcoxon-test importance</span></span>
<span><span class="va">fit.balanced</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/nestedcv/man/nestcv.train.html" class="external-link">nestcv.train</a></span><span class="op">(</span></span>
<span>  y <span class="op">=</span> <span class="va">synthetic_metadata</span><span class="op">$</span><span class="va">response</span>,</span>
<span>  x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">synthetic_rnaseqData</span><span class="op">)</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"rf"</span>,</span>
<span>  modifyX <span class="op">=</span> <span class="st">"multiDEGGs_combined_filter"</span>,</span>
<span>  modifyX_options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    filter_method <span class="op">=</span> <span class="st">"wilcoxon"</span>, </span>
<span>    nfilter <span class="op">=</span> <span class="fl">40</span>,</span>
<span>    dynamic_nfilter <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span>,</span>
<span>  modifyX_useY <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  n_outer_folds <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  n_inner_folds <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="available-statistical-methods">Available Statistical Methods<a class="anchor" aria-label="anchor" href="#available-statistical-methods"></a>
</h4>
<ul>
<li>
<strong><code>"ttest"</code></strong>: Two-sample t-test for
differential expression</li>
<li>
<strong><code>"wilcoxon"</code></strong>: Wilcoxon rank-sum test
(non-parametric alternative to t-test)</li>
<li>
<strong><code>"ranger"</code></strong>: Random Forest variable
importance scoring (the <code>ranger</code> package must be installed
first)</li>
<li>
<strong><code>"glmnet"</code></strong>: Elastic net regularization
coefficients</li>
<li>
<strong><code>"pls"</code></strong>: Partial Least Squares variable
importance</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="practical-considerations">Practical considerations<a class="anchor" aria-label="anchor" href="#practical-considerations"></a>
</h3>
<p>Before implementing multiDEGGs in your machine learning pipeline,
it’s highly recommended to first run a preliminary analysis on your
complete dataset to assess the number of differential interactions
detected. This exploratory step can guide your choice of approach and
parameter settings.</p>
<p>If multiDEGGs identifies only a small number of differential
interactions (e.g., fewer than 10-20 pairs), these features alone may
lack sufficient predictive power. In such cases, consider:</p>
<ul>
<li>Using <code><a href="../reference/multiDEGGs_combined_filter.html">multiDEGGs_combined_filter()</a></code> to integrate
network-based features with traditional statistical selection
methods</li>
<li>Setting <code>keep_single_genes = TRUE</code> in
<code><a href="../reference/multiDEGGs_filter.html">multiDEGGs_filter()</a></code> to include individual genes involved in
the differential pairs</li>
<li>Adjusting the <code>percentile_vector</code> or significance
thresholds in the initial multiDEGGs analysis to potentially capture
more interactions</li>
</ul>
<p>Conversely, if a large number of differential interactions are
detected, <code><a href="../reference/multiDEGGs_filter.html">multiDEGGs_filter()</a></code> alone may provide sufficient
feature diversity for effective model training.</p>
</div>
<div class="section level3">
<h3 id="feature-engineering-details">Feature Engineering Details<a class="anchor" aria-label="anchor" href="#feature-engineering-details"></a>
</h3>
<p>Both functions create ratio-based features from significant gene
pairs (Gene A / Gene B), which capture the relative expression
relationships that drive differential network connectivity. The
<code>predict</code> methods automatically handle the feature
transformation for both training and test data within each
cross-validation fold, ensuring no information leakage.</p>
<p><strong>Note:</strong> If no significant differential interactions
are found in a particular fold, both functions automatically fall back
to t-test-based selection to ensure robust performance across all
scenarios. This fallback is indicated by a printed “0” during
execution.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] caret_7.0-1      lattice_0.22-7   ggplot2_3.5.2    nestedcv_0.8.0  </span></span>
<span><span class="co">#&gt; [5] multiDEGGs_1.1.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1     timeDate_4041.110    dplyr_1.1.4         </span></span>
<span><span class="co">#&gt;  [4] farver_2.1.2         matrixTests_0.2.3    fastmap_1.2.0       </span></span>
<span><span class="co">#&gt;  [7] pROC_1.18.5          digest_0.6.37        rpart_4.1.24        </span></span>
<span><span class="co">#&gt; [10] timechange_0.3.0     lifecycle_1.0.4      survival_3.8-3      </span></span>
<span><span class="co">#&gt; [13] magrittr_2.0.3       compiler_4.5.1       rlang_1.1.6         </span></span>
<span><span class="co">#&gt; [16] sass_0.4.10          tools_4.5.1          yaml_2.3.10         </span></span>
<span><span class="co">#&gt; [19] data.table_1.17.8    knitr_1.50           htmlwidgets_1.6.4   </span></span>
<span><span class="co">#&gt; [22] plyr_1.8.9           RColorBrewer_1.1-3   withr_3.0.2         </span></span>
<span><span class="co">#&gt; [25] purrr_1.1.0          desc_1.4.3           nnet_7.3-20         </span></span>
<span><span class="co">#&gt; [28] grid_4.5.1           stats4_4.5.1         e1071_1.7-16        </span></span>
<span><span class="co">#&gt; [31] future_1.58.0        globals_0.18.0       scales_1.4.0        </span></span>
<span><span class="co">#&gt; [34] iterators_1.0.14     MASS_7.3-65          cli_3.6.5           </span></span>
<span><span class="co">#&gt; [37] rmarkdown_2.29       ragg_1.4.0           generics_0.1.4      </span></span>
<span><span class="co">#&gt; [40] RcppParallel_5.1.10  future.apply_1.20.0  reshape2_1.4.4      </span></span>
<span><span class="co">#&gt; [43] proxy_0.4-27         qvalue_2.40.0        cachem_1.1.0        </span></span>
<span><span class="co">#&gt; [46] stringr_1.5.1        splines_4.5.1        parallel_4.5.1      </span></span>
<span><span class="co">#&gt; [49] matrixStats_1.5.0    vctrs_0.6.5          hardhat_1.4.1       </span></span>
<span><span class="co">#&gt; [52] glmnet_4.1-10        Matrix_1.7-3         jsonlite_2.0.0      </span></span>
<span><span class="co">#&gt; [55] listenv_0.9.1        systemfonts_1.2.3    foreach_1.5.2       </span></span>
<span><span class="co">#&gt; [58] gower_1.0.2          jquerylib_0.1.4      recipes_1.3.1       </span></span>
<span><span class="co">#&gt; [61] glue_1.8.0           parallelly_1.45.1    pkgdown_2.1.3       </span></span>
<span><span class="co">#&gt; [64] codetools_0.2-20     shape_1.4.6.1        lubridate_1.9.4     </span></span>
<span><span class="co">#&gt; [67] stringi_1.8.7        gtable_0.3.6         sfsmisc_1.1-21      </span></span>
<span><span class="co">#&gt; [70] tibble_3.3.0         pillar_1.11.0        htmltools_0.5.8.1   </span></span>
<span><span class="co">#&gt; [73] randomForest_4.7-1.2 ipred_0.9-15         lava_1.8.1          </span></span>
<span><span class="co">#&gt; [76] R6_2.6.1             zigg_0.0.2           textshaping_1.0.1   </span></span>
<span><span class="co">#&gt; [79] doParallel_1.0.17    evaluate_1.0.4       Rfast_2.1.5.1       </span></span>
<span><span class="co">#&gt; [82] bslib_0.9.0          class_7.3-23         Rcpp_1.1.0          </span></span>
<span><span class="co">#&gt; [85] nlme_3.1-168         prodlim_2025.04.28   xfun_0.52           </span></span>
<span><span class="co">#&gt; [88] fs_1.6.6             pkgconfig_2.0.3      ModelMetrics_1.2.2.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"multiDEGGs"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'multiDEGGs' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Sciacca E, et al. (2023). "DEGGs: An R package with shiny app for the</span></span>
<span><span class="co">#&gt;   identification of differentially expressed gene-gene interactions in</span></span>
<span><span class="co">#&gt;   high-Throughput sequencing data." _Bioinformatics_, *39*, btad192.</span></span>
<span><span class="co">#&gt;   doi:10.1093/bioinformatics/btad192</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1093/bioinformatics/btad192&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {DEGGs: An R package with shiny app for the identification of differentially expressed gene-gene interactions in high-Throughput sequencing data},</span></span>
<span><span class="co">#&gt;     author = {Elisabetta Sciacca and {et al.}},</span></span>
<span><span class="co">#&gt;     journal = {Bioinformatics},</span></span>
<span><span class="co">#&gt;     year = {2023},</span></span>
<span><span class="co">#&gt;     volume = {39},</span></span>
<span><span class="co">#&gt;     pages = {btad192},</span></span>
<span><span class="co">#&gt;     doi = {10.1093/bioinformatics/btad192},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Elisabetta Sciacca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
